scripts/updateExtractions.js

```diff
@@ -0,0 +1,147 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const { execFileSync } = require("child_process");
+const yargs = require("yargs");
+const argv = yargs(process.argv).argv;
+
+// parse args
+const isCheck = "check" in argv;
+const targets = argv.targets;
+if (!targets) {
+  console.error(
+    `Fail! Please specify the "target" argument as a comma-separated list of paths to the target directories and files where extraction insertions need to be made.`,
+  );
+  process.exit(1);
+}
+
+// gather extractions from different packages into the workspace root
+console.log(`Gathering extractions from different packages...`);
+execFileSync("node", [path.join(__dirname, "gatherDocs.js")], { stdio: "inherit", cwd: path.join(__dirname, "..") });
+
+// set up constants
+const extractionsDir = path.join(__dirname, "..", "build/docs/extract");
+const extractionStart = "<!-- BEGIN EXTRACTION -->";
+const extractionEnd = "<!-- END EXTRACTION -->";
+const targetFileExtensions = [".ts", ".tsx", ".md"];
+const re = /^(\s*)(<!--|\/\/|\/\*)\s*\[\[include:\s*([\w\d\._-]+)(,[\s]*([\w\d_]+))?\]\]/;
+const reIndentIndex = 1;
+const reExtractionNameIndex = 3;
+const reExtractionTypeIndex = 5;
+
+const changedFiles = [];
+targets.split(",").forEach((target) => {
+  console.log(`Processing target "${target}"...`);
+  const targetStat = fs.lstatSync(target);
+  if (targetStat.isDirectory(target)) {
+    fs.readdirSync(target, { recursive: true }).forEach((fileName) => {
+      handleTargetFile(path.join(target, fileName));
+    });
+  } else if (targetStat.isFile(target)) {
+    handleTargetFile(target);
+  } else {
+    console.error(`Fail! Target "${target}" is not a valid directory or file.`);
+    process.exit(1);
+  }
+
+  if (isCheck) {
```

Comment:
What if code examples are modified, committed and not extracted? This doesn't seem to check that the extracted code matches the code in the documentation

Reply:
Not sure I understand what you're asking. 

You're not supposed to edit such snippets directly in the markdown - if you do that, the CI will fail as the snippet is going to be replaced with the extraction from compiled code. Then, since CI runs this script with `--check` flag, we're going to do `git status` and detect file modifications, which means the snippet in markdown doesn't match the extraction from compiled code.

Reply:
Oh, I see now, I thought that `git status` was done before the changes